packages/skeleton-svelte/src/lib/components/Rating/Rating.svelte

```diff
@@ -0,0 +1,125 @@
+<script lang="ts">
+	import type { RatingProps } from './types.js';
+
+	let {
+		value = $bindable(0),
+		max = 5,
+		interactive = false,
+		fraction = 1,
+		// Root
+		base = 'flex w-full',
+		text = 'text-token',
+		fill = 'fill-token',
+		justify = 'justify-center',
+		spaceX = 'space-x-2',
+		classes = '',
+		// Item ---
+		itemBase = 'w-full',
+		itemAspect = 'aspect-[1/1]',
+		itemClasses = '',
+		// Snippets ---
+		emptyIcon,
+		fullIcon
+	}: RatingProps = $props();
+
+	let hoverValue = $state(0);
+	let hovering = $state(false);
+	let lastSelectedFraction = $state(1);
+
+	function onRatingClick(event: Event, order: number) {
+		// handling mouse
+		if (event instanceof PointerEvent) {
+			const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+			console.log(event);
+			const fractionWidth = ratingRect.width / fraction;
+			const left = event.clientX - ratingRect.left;
+			lastSelectedFraction = Math.floor(left / fractionWidth) + 1;
+		}
+		// handling keyboard
+		else {
+		}
+		value = order + lastSelectedFraction / fraction;
+	}
+
+	function onRatingHover(event: Event, order: number) {
+		// handling mouse
+		if (event instanceof MouseEvent) {
+			const ratingRect = (event.currentTarget as HTMLElement).getBoundingClientRect();
+			const fractionWidth = ratingRect.width / fraction;
+			const left = event.clientX - ratingRect.left;
+			lastSelectedFraction = Math.floor(left / fractionWidth) + 1;
+		}
+		// handling keyboard
+		else {
```

Comment:
Looks like some logic is missing from the `else` condition?

Reply:
We agreed last time to handle keyboard logic after the first review, and I have some questions about how are we going to handle fractions with Keyboard, so I would wait until the first wave of changes are approved :+1:

Reply:
@endigo9740 Do you have any suggestions on how to handle the keyboard events with fractions? do we ignore the fractions or use something like arrow to set the different fractions?

Reply:
@Mahmoud-zino I like how Mantine handles this. Provide focus, then use left/right arrows:
https://mantine.dev/core/rating/#fractions